tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -477,25 +477,32 @@ module.exports = {
      * @returns {Array} array of sharing indicator
      */
     getShareIndicatorsForResource: async function(fileName, sharingIndicatorExpectedToBeVisible) {
-      const resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
-      const shareIndicatorsXpath =
-        resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
+      let resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
+      let shareIndicatorsXpath = resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
       const indicators = []
       await this.waitForFileVisible(fileName)
       const {
         waitForNegativeConditionTimeout,
         waitForConditionTimeout,
         waitForConditionPollInterval
       } = client.globals
-      await this.waitForElementVisible({
-        selector: shareIndicatorsXpath,
-        locateStrategy: this.elements.shareIndicatorsInFileRow.locateStrategy,
-        abortOnFailure: false,
-        timeout: sharingIndicatorExpectedToBeVisible
-          ? waitForConditionTimeout
-          : waitForNegativeConditionTimeout,
-        pollInterval: waitForConditionPollInterval
-      })
+      try {
+        await this.waitForElementVisible({
+          selector: shareIndicatorsXpath,
+          locateStrategy: this.elements.shareIndicatorsInFileRow.locateStrategy,
+          abortOnFailure: false,
+          timeout: sharingIndicatorExpectedToBeVisible
+            ? waitForConditionTimeout
+            : waitForNegativeConditionTimeout,
+          pollInterval: waitForConditionPollInterval
+        })
+      } catch (error) {
+        console.log(error)
+        console.log('\n Retrying with new reference \n')
+        resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
+        shareIndicatorsXpath = resourceRowXpath + this.elements.shareIndicatorsInFileRow.selector
```

Comment:
the value of these variables does not seems to be different from the initial value. Why do you need to update it here?

Reply:
During `await this.waitForElementVisible({` line error is thrown `StaleElementReferenceException`. So I put that code inside try-catch and retry it.